src/quad/CMakeLists.txt

```diff
@@ -0,0 +1,152 @@
+    list(APPEND SLEEF_HEADER_COMMANDS COMMAND $<TARGET_FILE:qmkrename> ${QUAD_HEADER_PARAMS_${SIMD}} >> ${SLEEFQUAD_INCLUDE_HEADER})
+  endif()
+endforeach()
+
+if(MSVC)
+  string(REPLACE "/" "\\" sleef_footer_input_file "${SLEEFQUAD_ORG_FOOTER}")
+  list(APPEND SLEEF_HEADER_COMMANDS COMMAND type ${sleef_footer_input_file} >> ${SLEEFQUAD_INCLUDE_HEADER})
+else()
+  list(APPEND SLEEF_HEADER_COMMANDS COMMAND cat ${SLEEFQUAD_ORG_FOOTER} >> ${SLEEFQUAD_INCLUDE_HEADER})
+endif()
+
+add_custom_command(OUTPUT ${SLEEFQUAD_INCLUDE_HEADER}
+  ${SLEEF_HEADER_COMMANDS}
+  DEPENDS
+    ${SLEEFQUAD_ORG_HEADER}
+    ${SLEEFQUAD_ORG_FOOTER}
+    qmkrename
+)
+
+# --------------------------------------------------------------------
+# qmkrename
+# qrenameXXX.h for each vector extension
+# --------------------------------------------------------------------
+# Helper executable: generates parts of the sleef header file
+add_host_executable(qmkrename qmkrename.c)
+set_target_properties(qmkrename PROPERTIES ${COMMON_TARGET_PROPERTIES})
+
+set(HEADER_FILES_GENERATED "")
+foreach(SIMD ${SLEEFQUAD_SUPPORTED_EXT})
+  if(COMPILER_SUPPORTS_${SIMD})
+    string(TOLOWER ${SIMD} SIMDLC)
+    set(HEADER_${SIMD} ${CMAKE_CURRENT_BINARY_DIR}/include/qrename${SIMDLC}.h)
+    list(APPEND HEADER_FILES_GENERATED ${HEADER_${SIMD}})
+
+    # Generate qmkrename commands
+    add_custom_command(OUTPUT ${HEADER_${SIMD}}
+      COMMAND echo Generating qrename${vecarch}.h: qmkrename ${QUAD_RENAME_PARAMS_${SIMD}}
+      COMMAND $<TARGET_FILE:qmkrename> ${QUAD_RENAME_PARAMS_${SIMD}} > ${HEADER_${SIMD}}
+      DEPENDS qmkrename
+    )
+    add_custom_target(qrename${SIMD}.h_generated DEPENDS ${HEADER_${SIMD}})
+  endif()
+endforeach()
+
+# --------------------------------------------------------------------
+# sleefquad_headers
+# --------------------------------------------------------------------
+add_custom_target(sleefquad_headers ALL
+  DEPENDS
+    ${SLEEFQUAD_INCLUDE_HEADER}
+    ${HEADER_FILES_GENERATED}
+)
+
+# --------------------------------------------------------------------
+# libsleefquad
+# --------------------------------------------------------------------
+
+foreach(SIMD ${SLEEFQUAD_SUPPORTED_EXT})
+  if(COMPILER_SUPPORTS_${SIMD})
+    string(TOLOWER ${SIMD} SIMDLC)
+    set(OBJECT "sleefquad${SIMDLC}_obj")
+    add_library(${OBJECT} OBJECT sleefsimdqp.c ${HEADER_${SIMD}})    
+
+    if(COMPILER_SUPPORTS_BUILTIN_MATH)
+      target_compile_definitions(${OBJECT} PRIVATE ENABLE_BUILTIN_MATH=1)
+    endif()
+    target_compile_definitions(${OBJECT} PRIVATE ENABLE_${SIMD}=1 DORENAME=1 ${COMMON_TARGET_DEFINITIONS})
+
+    set_target_properties(${OBJECT} PROPERTIES ${COMMON_TARGET_PROPERTIES})
+    add_dependencies(${OBJECT} qrename${SIMD}.h_generated)
+    target_compile_options(${OBJECT} PRIVATE ${FLAGS_ENABLE_${SIMD}})
+
+    list(APPEND SLEEFQUAD_OBJECTS $<TARGET_OBJECTS:${OBJECT}>)
+  endif()
+endforeach()
+
+add_library(sleefquad ${SLEEFQUAD_OBJECTS})
```

Comment:
Is there any reason why this is built as a separate library and not as part of libsleef?